### PR TITLE
[extension/jaegerremotesampling] remove misleading agent endpoint

### LINKED ISSUE #18058: [extension/jaegerremotesampling] Misleading implementation on two endpoints
### Component(s)

extension/jaegerremotesampling

## Description

This extension serves the same output on two HTTP endpoints `/` and `/sampling`, which presumably was meant to emulate the Jaeger Agent's behavior. However, in the Jaeger Agent the `/` endpoint returns a legacy format (see https://www.jaegertracing.io/docs/1.41/apis/#remote-sampling-configuration-stable).

This difference either needs to be called out in the README or fixed (although returning legacy format from this extension does not make sense, I'd rather it did not support this endpoint).

### Collector version

0.70.0

## Environment
any

### OpenTelemetry Collector configuration

_No response_

### Log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by frzifus**
**Description:** <Describe what has changed.>

Removed `/` http endpoint from remotesampling extension.

**Link to tracking Issue:** 
- Closes #18058

### CODE REVIEW

**Review by jpkrohling — APPROVED** [review]

**Review by codeboten — COMMENTED** [review]
Is this not a breaking change?

**Comment by jpkrohling** [comment]
The current behavior is not in line with what Jaeger does, so, no current Jaeger clients work with this. But you are right, @codeboten, this is technically a breaking change.

Should we do this in two phases? First deprecate this, issuing a warn in the logs in case this endpoint gets called, removing it in N+2?

**Review by codeboten — COMMENTED** [review]
> so, no current Jaeger clients work with this

If that's the case I think it's probably ok to mark this as a breaking change (in case anyone out there is using this) and remove it in the same change.

File: .chloggen/remove_misleading_jaegerremotesampling_endpoint.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
```
Comment (jpkrohling) [inline .chloggen/remove_misleading_jaegerremotesampling_endpoint.yaml:2]: As per @codeboten's comment, this should be "breaking change".

**Review by jpkrohling — CHANGES_REQUESTED** [review]

File: .chloggen/remove_misleading_jaegerremotesampling_endpoint.yaml (hunk 1 above)
Comment (frzifus) [reply .chloggen/remove_misleading_jaegerremotesampling_endpoint.yaml:2]:
```suggestion
change_type: breaking
```

**Review by codeboten — APPROVED** [review]

**Review by MovieStoreGuy — APPROVED** [review]

**Review by jpkrohling — APPROVED** [review]

File: extension/jaegerremotesampling/internal/http.go [hunk 2]
```diff
@@ -58,9 +58,6 @@ func NewHTTP(telemetry component.TelemetrySettings, settings confighttp.HTTPServ
 	}
 
 	srv.mux = http.NewServeMux()
-	// the legacy endpoint
-	srv.mux.Handle("/", http.HandlerFunc(srv.samplingStrategyHandler))
-
 	// the new endpoint -- not strictly necessary, as the previous one would match it
```
Comment (yurishkuro) [inline extension/jaegerremotesampling/internal/http.go:61]: may want to fix the comment: "the previous one" doesn't exist anymore

Comment (frzifus) [reply extension/jaegerremotesampling/internal/http.go:61]: Ups, good catch :X

**Pull request merged**